src/main/webapp/app/guided-tour/guided-tour.service.ts

```diff
@@ -0,0 +1,367 @@
+        document.body.classList.add('tour-open');
+    }
+
+    /**
+     * Define first and last tour step based on amount of tour steps
+     * @private
+     */
+    private _setFirstAndLast(): void {
+        if (this._currentTour) {
+            this._onLastStep = this._currentTour.steps.length - 1 === this._currentTourStepIndex;
+            this._onFirstStep = this._currentTourStepIndex === 0;
+        }
+    }
+
+    /* Check if highlighted element is available */
+    private _checkSelectorValidity(): boolean {
+        if (this._currentTour) {
+            if (this._currentTour.steps[this._currentTourStepIndex].selector) {
+                const selectedElement = document.querySelector(this._currentTour.steps[this._currentTourStepIndex].selector!);
+                if (!selectedElement) {
+                    this.errorHandler.handleError(
+                        // If error handler is configured this should not block the browser.
+                        new Error(
+                            `Error finding selector ${this._currentTour.steps[this._currentTourStepIndex].selector} on step ${this._currentTourStepIndex + 1} during guided tour: ${
+                                this._currentTour.settingsId
+                            }`,
+                        ),
+                    );
+                    return false;
+                }
+            }
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     *  Is last tour step
+     */
+    public get onLastStep(): boolean {
+        return this._onLastStep;
+    }
+
+    /**
+     * Is first tour step
+     */
+    public get onFirstStep(): boolean {
+        return this._onFirstStep;
+    }
+
+    /* Show resize message */
+    public get onResizeMessage(): boolean {
+        return this._onResizeMessage;
+    }
+
+    /* Current tour step number */
+    public get currentTourStepDisplay(): number {
+        return this._currentTourStepIndex + 1;
+    }
+
+    /* Total count of tour steps */
+    public get currentTourStepCount(): any {
+        return this._currentTour && this._currentTour.steps ? this._currentTour.steps.length : 0;
+    }
+
+    /* Prevents the tour from advancing by clicking the backdrop */
+    public get preventBackdropFromAdvancing(): boolean {
+        if (this._currentTour) {
+            return this._currentTour && (this._currentTour.preventBackdropFromAdvancing ? this._currentTour.preventBackdropFromAdvancing : false);
+        }
+        return false;
+    }
+
+    /**
+     * Get the tour step with defined orientation
+     * @param index current tour step index
+     */
+    private getPreparedTourStep(index: number): TourStep | undefined {
+        if (this._currentTour) {
+            return this.setTourOrientation(this._currentTour.steps[index]);
+        } else {
+            return undefined;
+        }
+    }
+
+    /**
+     * Set orientation of the passed on tour step
```

Comment:
This JSDoc is very light for such a heavy method.
I wonder in general why there is so much going on this method:
- Why can the orientation be a string?
- What happens if the window doesn't have a certain width?